File: backend/src/models/Payment.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

// Pago: aporte realizado por un miembro hacia la meta
const Payment = sequelize.define('Payment', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  amount: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false,
  },
  note: {
    type: DataTypes.STRING,
    allowNull: true,
  },
  paymentDate: {
    type: DataTypes.DATEONLY,
    allowNull: false,
    defaultValue: DataTypes.NOW,
  },
  // memberId y goalId se agregan por las asociaciones
}, {
  tableName: 'payments',
  timestamps: true,
});

module.exports = Payment;

File: frontend/app/src/main/java/com/taller/parcial2/model/Payment.kt
package com.taller.parcial2.model

// Modelo: pago realizado por un miembro
data class Payment(
    val id: String,
    val amount: Double,
    val note: String?,
    val paymentDate: String,
    val memberId: String,
    val memberName: String?,
    val goalId: String,
    val createdAt: String?
)
